### PR TITLE
Adding Universal Pages in FoodAdvisor Frontend

### PR DESCRIPTION
**Pull request by malgamves**
In this PR, I add the following functionality, 

- [x]  🖥 On the frontend side, manage the fact that pages can be created on the fly
*Be able to render any pages*

#### Pending 
- [x]  🖥 Update the frontend footer to fetch the data from the Footer single type

### CODE REVIEW

File: client/src/containers/App/index.js [hunk 1]
```diff
@@ -44,6 +45,7 @@ function App() {
         <Route path="/about" component={About} exact />
         <Route path="/blog" component={Blog} exact />
         <Route path="/:id/:content" component={RestaurantPage} exact />
+        <Route path="/:title" component={UniversalPage} exact />
```
Comment (soupette) [inline client/src/containers/App/index.js:48]: Can you use `slug` instead of `title` ?

**Review by soupette — CHANGES_REQUESTED** [review]

**Comment by malgamves** [comment]
Working on the changes! Thanks for the tips

File: client/src/containers/UniversalPage/tests/index.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+// import { enzymeFind } from 'styled-components/test-utils';
+
+import UniversalPage from '../index';
+
+describe('<UniversalPage />', () => {
+  it('should not crash', () => {
+    shallow(<UniversalPage />);
```
[the remark this answers is not in the document]
Comment (malgamves) [reply client/src/containers/UniversalPage/tests/index.test.js:9]: sure thing, changing the `title` to `slug` also

File: api/components/footer/columns.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "collectionName": "components_footer_columns",
+  "info": {
+    "name": "Columns",
+    "icon": "archway",
+    "description": ""
+  },
+  "options": {},
+  "attributes": {
+    "link": {
```
[the remark this answers is not in the document]
Comment (Mcastres) [reply api/components/footer/columns.json]: LGTM

File: client/src/components/FooterColumn/tests/index.test.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// import React from 'react';
```
[the remark this answers is not in the document]
Comment (Mcastres) [reply client/src/components/FooterColumn/tests/index.test.js]: LGTM

File: client/src/queries/footer.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable import/prefer-default-export */
+import gql from 'graphql-tag';
+
+const GET_FOOTER = gql`
+query newFooter {
```
[the remark this answers is not in the document]
Comment (Mcastres) [reply client/src/queries/footer.js]: LGTM

File: client/src/components/FooterColumn/index.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    <Nav navbar>
+      {columns.map(column => (
+        <Column links={column.link} key={column.link.label} />
+      ))}
+    </Nav>
+  );
+}
+
+FooterColumn.defaultProps = {};
+FooterColumn.propTypes = {};
```
[the remark this answers is not in the document]
Comment (Mcastres) [reply client/src/components/FooterColumn/index.js]: LGTM

File: client/src/components/FooterColumn/Column.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+          >
+            <span title={link.name}>{link.name}</span>
+          </Link>
+        );
+      })}
+    </div>
+  );
+};
+
+export default Column;
```
[the remark this answers is not in the document]
Comment (Mcastres) [reply client/src/components/FooterColumn/Column.js:50]: LGTM

File: client/src/components/Footer/tests/index.test.js [hunk 8]
```diff
@@ -1,54 +1,9 @@
...
-  });
-
-  it('should use the defaultProps', () => {
-    const {
-      defaultProps: { onSubmit },
-    } = Footer;
-
-    expect(onSubmit).toBeDefined();
-    expect(onSubmit({ preventDefault: jest.fn() })).toBe(undefined);
+    shallow(<Footer match={{ params: { columns: [] } }} />);
```
[the remark this answers is not in the document]
Comment (Mcastres) [reply client/src/components/Footer/tests/index.test.js:7]: Deleted, LGTM

File: client/src/components/Footer/index.js [hunk 9]
```diff
@@ -71,12 +52,10 @@ function Footer({ links, onSubmit }) {
 }
 
 Footer.defaultProps = {
-  links: [],
-  onSubmit: e => e.preventDefault()
+  onSubmit: e => e.preventDefault(),
```
[the remark this answers is not in the document]
Comment (Mcastres) [reply client/src/components/Footer/index.js:56]: LGTM

[the remark this answers is not in the document]
Comment (Mcastres) [reply client/src/components/Footer/index.js:56]: LGTM

**Review by Mcastres — APPROVED** [review]
LGTM

**Review by Mcastres — APPROVED** [review]
LGTM

File: client/src/components/Footer/index.js [hunk 10]
```diff
@@ -57,6 +57,7 @@ Footer.defaultProps = {
 };
 Footer.propTypes = {
   onSubmit: PropTypes.func,
+  columns: PropTypes.array,
```
Comment (soupette) [inline client/src/components/Footer/index.js:60]: `columns: PropTypes.array.isRequired`

File: client/src/components/FooterColumn/index.js [hunk 11]
```diff
@@ -30,13 +31,15 @@ function FooterColumn({ columns }) {
...
-        <Column links={column.links} key={key} />
+        <Column links={column.links} key={key.id} />
       ))}
     </Nav>
   );
 }
 
-FooterColumn.defaultProps = {};
+FooterColumn.defaultProps = {
+  column:[],
```
Comment (soupette) [inline client/src/components/FooterColumn/index.js:41]: This should be `columns`

**Review by soupette — CHANGES_REQUESTED** [review]

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**